lighty-applications/lighty-rcgnmi-app-aggregator/lighty-rcgnmi-app/src/main/java/io/lighty/applications/rcgnmi/app/RCgNMIApp.java

```diff
@@ -92,18 +100,20 @@ public void start(final String[] args) {
         try {
             final boolean hasStarted = rcgnmiLightyModule.start().get();
             if (hasStarted) {
-                // Register shutdown hook for graceful shutdown
                 LOG.info("Registering ShutdownHook to gracefully shutdown application");
                 registerShutdownHook(rcgnmiLightyModule);
                 LOG.info("RCgNMI lighty.io application started in {}", stopwatch.stop());
             } else {
-                LOG.error("Unable to start RCgNMI lighty.io application!");
+                stop();
```

Comment:
Do not call `stop` here in each error state. Just throw exception and handle it one layer higher (in `main` method in this case.